hardware/arduino/avr/cores/arduino/WString.cpp

```diff
@@ -189,25 +188,22 @@ String & String::copy(const __FlashStringHelper *pstr, unsigned int length)
 	return *this;
 }
 
-#if __cplusplus >= 201103L || defined(__GXX_EXPERIMENTAL_CXX0X__)
+#if STRING_HAS_MOVE
 void String::move(String &rhs)
 {
-	if (buffer) {
-		if (rhs && capacity >= rhs.len) {
-			strcpy(buffer, rhs.buffer);
-			len = rhs.len;
-			rhs.len = 0;
-			return;
-		} else {
+	if (this != &rhs)
+	{
+		if (buffer != NULL)
 			free(buffer);
```

Comment:
FYI: The `if` is redundant since `free(nullptr)` is defined to be a safe no-op.
http://www.cplusplus.com/reference/cstdlib/free/

Reply:
Thanks. Removed the null checks before free.